src/system/file_path.hh

```diff
@@ -19,6 +19,7 @@
 #define FILE_NAME_HH_
 
 #include <string>
+#include <boost/filesystem.hpp>
```

Comment:
We have to avoid this include in the header. Move implementation to the cc file, even for the opeartor, inlining is not necessary here.